core/src/stack/inputs.rs

```diff
@@ -16,17 +16,33 @@ pub struct StackInputs {
 }
 
 impl StackInputs {
+    // CONSTANTS
+    // --------------------------------------------------------------------------------------------
+
+    pub const MAX_STACK_INPUTS_SIZE: usize = u16::MAX as usize;
+
     // CONSTRUCTORS
     // --------------------------------------------------------------------------------------------
 
     /// Returns [StackInputs] from a list of values, reversing them into a stack.
-    pub fn new(mut values: Vec<Felt>) -> Self {
+    ///
+    /// # Errors
+    /// Returns an error if the number of input values exceeds the allowed maximum.
+    pub fn new(mut values: Vec<Felt>) -> Result<Self, InputError> {
+        if values.len() > Self::MAX_STACK_INPUTS_SIZE {
+            return Err(InputError::InputLengthExceeded(Self::MAX_STACK_INPUTS_SIZE, values.len()));
+        }
         values.reverse();
-        Self { values }
+
+        Ok(Self { values })
     }
 
-    /// Attempts to create stack inputs from an iterator of numbers, failing if they do not
-    /// represent a valid field element.
+    /// Attempts to create stack inputs from an iterator of integers.
+    ///
+    /// # Errors
+    /// Returns an error if:
+    /// - The values do not represent a valid field element.
+    /// - Number of values in the iterator exceeds the allowed maximum number of input values.
     pub fn try_from_values<I>(iter: I) -> Result<Self, InputError>
```

Comment:
I just noticed that `StackOutputs` has `try_from_ints()` method. So, we should either name this one `try_from_ints()` or rename the other one into `try_from_values()`. I think the first option is probably a bit better.